src/en/kunmangato/src/eu/kanade/tachiyomi/extension/en/kunmangato/KunMangaTo.kt

```diff
@@ -0,0 +1,267 @@
+        val form = FormBody.Builder()
+            .add(
+                "chapterIdx",
+                response.request.url.toString().substringBefore(".html").substringAfterLast("-"),
+            )
+            .build()
+        val chaptersRequest = POST(
+            "$baseUrl/chapter-resources",
+            Headers.headersOf(
+                "X-CSRF-TOKEN",
+                document.select("head > meta[name=\"csrf-token\"]").first()!!.attr("content"),
+                "Cookie",
+                response.headers.values("set-cookie")
+                    .find { value -> value.startsWith("kunmanga_session") }!!,
+            ),
+            form,
+        )
+        val chaptersResponse = client.newCall(chaptersRequest).execute()
+        val chapterResources =
+            json.decodeFromString<KunMangaToChapterResourcesDto>(chaptersResponse.body.string())
+        return chapterResources.data.resources.map { resource ->
+            Page(
+                resource.id,
+                "",
+                resource.thumb,
+            )
+        }
+    }
+
+    private val filterNames =
+        arrayOf("manga_genre_id" to "Genre", "manga_type_id" to "Type", "status" to "Status")
+
+    private var filterValues: Map<Pair<String, String>, List<Pair<String, String>>>? = null
+
+    private var fetchFilterValuesAttempts: Int = 0
+
+    private fun parseFilters(document: Document): Map<Pair<String, String>, List<Pair<String, String>>> {
+        return filterNames.associateBy(
+            { it },
+            {
+                document.select("[name=\"${it.first}\"] option")
+                    .map { option -> Pair(option.attr("value"), option.text()) }
+            },
+        )
+    }
+
+    private fun fetchFiltersValues() {
+        if (fetchFilterValuesAttempts < 3 && filterValues == null) {
+            thread {
+                try {
+                    filterValues = parseFilters(
+                        client.newCall(searchMangaRequest(1, "", FilterList())).execute().asJsoup(),
+                    )
+                } finally {
+                    fetchFilterValuesAttempts++
+                }
+            }
+        }
+    }
+
+    override fun getFilterList(): FilterList {
+        fetchFiltersValues()
+        return if (filterValues != null) {
+            FilterList(
+                filterValues!!.map { filterValue ->
+                    UriPartFilter(
+                        filterValue.key.second,
+                        filterValue.key.first,
+                        filterValue.value,
+                    )
+                },
+            )
+        } else {
+            FilterList(Filter.Header("Press 'Reset' to attempt to fetch the filters"))
+        }
+    }
+
+    override fun imageUrlParse(document: Document): String {
+        throw UnsupportedOperationException()
+    }
+
+    override fun chapterListSelector(): String {
+        throw Exception("Not used")
+    }
```

Comment:
```suggestion
    override fun chapterListSelector(): String = ".chapter-item"
```
and use the value in `chapterListParse`

Reply:
Like this, even though technically it's an id and not a CSS selector?

> document.getElementById(chapterListSelector())!!.attr(...

Reply:
i mean in 
```kotlin
val chapterItems = chaptersDocument.select(".chapter-item")
```

you can use this for the id, but `getElementById` is fine
```kotlin
 document.selectFirst("#chapterList")
```

Reply:
Fixed, I guess